scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -0,0 +1,390 @@
+                extractArchive(saveFilePath);
+            }
+        } // If it is a folder, download all its contents recursively
+        else {
+            ftpClient.changeWorkingDirectory(ftpRemoteFilePath);
+            downloadDirectory(ftpClient, "", "", ftpLocalFilePath);
+        }
+        getOut().println("END Import file(s) from FTP.");
+    }
+
+    private void extractArchive(String saveFilePath) throws IOException {
+        try (ZipFile zipFile = new ZipFile(saveFilePath)) {
+            // Create an enumeration of the entries in the zip file
+            Enumeration zipFileEntries = zipFile.entries();
+            // Process each entry
+            while (zipFileEntries.hasMoreElements()) {
+                // grab a zip file entry
+                ZipEntry entry = (ZipEntry) zipFileEntries.nextElement();
+                Path path = Paths.get(ftpLocalFilePath, entry.getName());
+                getOut().println("path: " + path.toString());
+                if (entry.isDirectory()) {
+                    Files.createDirectories(path);
+                } else {
+                    Path parentDir = path.getParent();
+                    getOut().println("parentDir of path: " + parentDir.toString());
+                    if (!Files.exists(parentDir)) {
+                        Files.createDirectories(parentDir);
+                    }
+                    if (Files.exists(path)) {
+                        new File(path.toString()).delete();
+                    }
+
+                    Files.copy(zipFile.getInputStream(entry), path);
+                }
+            }
+        }
+    }
+
+    private void FtpPut(FTPClient ftpClient) throws IOException, IllegalArgumentException {
+        if (ftpLocalFilePath.isEmpty()) {
+            throw new IllegalArgumentException("The ftp local file path variable is empty. Please, enter a value.");
+        }
+        getOut().println("Exporting file(s) from " + ftpLocalFilePath + " to " + ftpRemoteFilePath);
+        File localFile = new File(ftpLocalFilePath);
+        // if the ftp remote file path does not exist then it is created.
+        if (!checkDirectoryExists(ftpClient, ftpRemoteFilePath)) {
+            ftpClient.makeDirectory(ftpRemoteFilePath);
+            getOut().println(ftpRemoteFilePath + " does not exist. Therefore, it is created.");
+        } else {
+            getOut().println(ftpRemoteFilePath + " exists.");
+        }
+        // If it is a single file:
+        if (localFile.isFile()) {
+            getOut().println("A single FILE to upload");
+            ftpClient.changeWorkingDirectory(ftpRemoteFilePath);
+            boolean uploaded = uploadSingleFile(ftpClient, ftpLocalFilePath, localFile.getName());
+            if (uploaded) {
+                getOut().println("UPLOADED a file to: " + Paths.get(ftpRemoteFilePath, localFile.getName()));
+            } else {
+                getOut().println("COULD NOT upload the file: " + ftpLocalFilePath);
```

Comment:
this should be an error message, please use getErr().

In that case, it seems strange that the status of the task is not faulty, and that it silently terminates

Reply:
done